demos/python_demos/im2latex_demo/utils.py

```diff
@@ -0,0 +1,45 @@
+import pickle as pkl
+import json
+
+START_TOKEN = 0
+PAD_TOKEN = 1
+END_TOKEN = 2
+UNK_TOKEN = 3
+
+class Vocab():
+    """Vocabulary class which helps to get
+    human readable formula from sequence of integer tokens
+
+    Raises:
+        ValueError: If wrong extension file passed as 'vocab_path'
+    """
+    def __init__(self, vocab_path):
+        if '.pkl' in vocab_path:
+            with open(vocab_path, "rb") as f:
+                vocab_dict = pkl.load(f)
+        elif 'json' in vocab_path:
+            with open(vocab_path, "r") as f:
+                vocab_dict = json.load(f)
+                for k, v in vocab_dict['id2sign'].items():
+                    del vocab_dict['id2sign'][k]
+                    vocab_dict['id2sign'][int(k)] = v
```

Comment:
I'm not sure it's safe to delete elements from the dictionary while you're iterating over it. I would suggest to recreate the dict instead:

```python
vocab_dict['id2sign'] = {int(k): v for k, v in vocab_dict['id2sign'].items()}
```

Reply:
Done, thanks